File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/ICesiumPathModeValuePropertyWriter.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import javax.annotation.Nonnull;

/**
 * A writer that can write a value as a path mode.
 */
@SuppressWarnings({
    "unused",
    "deprecation",
    "serial"
})
public interface ICesiumPathModeValuePropertyWriter extends ICesiumPropertyWriter {
    /**
    * Writes the value as a path mode.
    * @param value The path mode.
    */
    void writePathMode(@Nonnull CesiumPathMode value);
}

File: DotNet/CesiumLanguageWriter/Advanced/ICesiumPathModeValuePropertyWriter.cs
namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// A writer that can write a value as a path mode.
    /// </summary>
    public interface ICesiumPathModeValuePropertyWriter : ICesiumPropertyWriter
    {
        /// <summary>
        /// Writes the value as a path mode.
        /// </summary>
        /// <param name="value">The path mode.</param>
        void WritePathMode(CesiumPathMode value);
    }
}
